numpy/lib/arraysetops.py

```diff
@@ -254,42 +310,98 @@ def reshape_uniq(uniq):
         uniq = np.swapaxes(uniq, 0, axis)
         return uniq
 
+    consolidated_base = np.array_view_root(consolidated)
+    orig_base = np.array_view_root(orig_ar)
+    if return_index or return_inverse:
+        # We won't perform an in-place sort even if requested
+        mask_is_sorted = assume_sorted
```

Comment:
Should this set `sort_inplace = False`?

Reply:
Almost, if we sort in-place oportunistically on a copy, `mask_is_sorted != assume_sorted or sort_inplace`. But some simplification of the sort is doable, just pushed that.